lib/protocol/redis/methods/sorted_sets.rb

```diff
@@ -68,6 +68,24 @@ def zrange(key, start, stop, with_scores: false)
 					
 					call("ZRANGE", key, *arguments)
 				end
+
+				# Return a range of members in a sorted set, by score. O(log(N)+M) with N being the number of elements in the sorted set and M the number of elements being returned. If M is constant (e.g. always asking for the first 10 elements with LIMIT), you can consider it O(log(N))..
+        # @see https://redis.io/commands/zrangebyscore
+        # @param key [Key]
+        # @param min [Integer]
+        # @param max [Integer]
+        # @param with_scores [Boolean] Return the scores of the elements together with the elements.
+        # @param limit [Array] Can be used to only get a range of the matching elements (similar to SELECT LIMIT offset, count in SQL).
+        # @example Retrieve the first 10 members with score `>= 0` and `<= 100`
+        #   redis.zrangebyscore("zset", "0", "100", :limit => [0, 10])
+        def zrangebyscore(key, min, max, with_scores: false, limit: nil)
+          arguments = [min, max]
+
+          arguments << 'WITHSCORES' if with_scores
+          arguments.concat(['LIMIT'] + limit) if limit
```

Comment:
Do you need to construct a temporary array here?

Reply:
@ioquatix I've pushed changes without constructing a temporary array.

Reply:
Why not just:

```
arguments.push('LIMIT', limit)
```

Reply:
Or if limit is an array, 

```
arguments.push('LIMIT', *limit)
```

Reply:
@ioquatix Good idea. I've also updated other methods in the file in order to be consistent.

Reply:
@ioquatix Let's merge it? 🚀